packages/firestore/ios/RNFBFirestore/RNFBFirestoreCollectionModule.m

```diff
@@ -216,6 +216,77 @@ - (void)invalidate {
                  }];
 }
 
+RCT_EXPORT_METHOD(aggregateQuery
+                  : (FIRApp *)firebaseApp
+                  : (NSString *)databaseId
+                  : (NSString *)path
+                  : (NSString *)type
+                  : (NSArray *)filters
+                  : (NSArray *)orders
+                  : (NSDictionary *)options
+                  : (NSArray *)aggregateQueries
+                  : (RCTPromiseResolveBlock)resolve
+                  : (RCTPromiseRejectBlock)reject) {
+  FIRFirestore *firestore = [RNFBFirestoreCommon getFirestoreForApp:firebaseApp
+                                                         databaseId:databaseId];
+  FIRQuery *query = [RNFBFirestoreCommon getQueryForFirestore:firestore path:path type:type];
+
+  NSMutableArray<FIRAggregateField *> *aggregateFields =
+      [[NSMutableArray<FIRAggregateField *> alloc] init];
+
+  for (NSDictionary *aggregateQuery in aggregateQueries) {
+    NSString *aggregateType = aggregateQuery[@"aggregateType"];
+    NSString *fieldPath = aggregateQuery[@"field"];
+
+    if ([aggregateType isEqualToString:@"count"]) {
+      [aggregateFields addObject:[FIRAggregateField aggregateFieldForCount]];
+    } else if ([aggregateType isEqualToString:@"sum"]) {
+      [aggregateFields addObject:[FIRAggregateField aggregateFieldForSumOfField:fieldPath]];
+    } else if ([aggregateType isEqualToString:@"average"]) {
+      [aggregateFields addObject:[FIRAggregateField aggregateFieldForAverageOfField:fieldPath]];
+    } else {
+      NSString *reason = [@"Invalid Aggregate Type: " stringByAppendingString:aggregateType];
+      @throw [NSException exceptionWithName:@"RNFB Firestore: Invalid Aggregate Type"
+                                     reason:reason
+                                   userInfo:nil];
```

Comment:
throwing in this way creates a different error handling pathway in the javascript as it bubbles up from native through to javascript to app if I recall correctly?

I think it would be best to reject in this case so that all flow is handled in the typical resolve/reject style and it's a Promise rejection vs a separate thing you have to try/catch

You can I think take the exception you create here and just inline it as you send it in to the common rejection method e.g. `[RNFBFirestoreCommon promiseRejectFirestoreException:reject error:error];`

```suggestion
      [RNFBFirestoreCommon promiseRejectFirestoreException:reject error:[NSException exceptionWithName:@"RNFB Firestore: Invalid Aggregate Type"
                                     reason:reason
                                   userInfo:nil]];
```

(hopefully that compiles or is close to compiling, and then yarn lint:ios:fix because it looks awful in this textbox)